lib/src/chart/pie_chart/pie_chart_painter.dart

```diff
@@ -93,6 +97,37 @@ class PieChartPainter extends BaseChartPainter<PieChartData> {
 
     var tempAngle = data.startDegreeOffset;
 
+    /// Paint the shadows
+    for (var i = 0; i < data.sections.length; i++) {
+      final section = data.sections[i];
+      if (section.shadow == null) continue;
+      final sectionDegree = sectionsAngle[i];
+
+      if (sectionDegree == 360) {
+        drawSectionFullCircleShadow(
+          section,
+          center,
+          centerRadius,
+          canvasWrapper,
+        );
+        break;
+      }
+
+      final sectionPath = generateSectionPath(
```

Comment:
Please store this variable in the outer scope to make it possible to re-use it on line 168 when we want to draw actual paths to improve the performance.